File: src/main/java/me/shiiyuko/manosaba/Manosaba.java
package me.shiiyuko.manosaba;

import com.mojang.logging.LogUtils;
import me.shiiyuko.manosaba.config.ManosabaConfig;
import me.shiiyuko.manosaba.config.ManosabaConfigScreen;
import me.shiiyuko.manosaba.init.ManosabaSounds;
import net.minecraft.client.Minecraft;
import net.minecraftforge.api.distmarker.Dist;
import net.minecraftforge.client.event.ClientPlayerNetworkEvent;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.eventbus.api.IEventBus;
import net.minecraftforge.eventbus.api.SubscribeEvent;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
import net.minecraftforge.fml.loading.FMLEnvironment;
import org.slf4j.Logger;

@Mod(Manosaba.MODID)
public class Manosaba {

    public static final String MODID = "manosaba";
    public static final Logger LOGGER = LogUtils.getLogger();

    /** 启动 logo 序列是否已播放（会话内仅一次：首次进主界面时播放）。 */
    public static boolean bootSequencePlayed = false;

    /** 标题 BGM 是否正在播放（ManosabaTitleScreen 自行管理）。
     *  为 true 时 MusicManager 被抑制（含子界面期间，防止其 5s 后重启音乐造成叠加）。 */
    public static boolean titleMusicPlaying = false;

    /** 进入世界：停止标题 BGM，MusicManager 恢复游戏音乐。 */
    @SubscribeEvent
    public void onPlayerLoggedIn(ClientPlayerNetworkEvent.LoggingIn event) {
        titleMusicPlaying = false;
        Minecraft.getInstance().getSoundManager().stop();
    }

    public Manosaba() {
        ManosabaConfig.register();
        IEventBus modEventBus = FMLJavaModLoadingContext.get().getModEventBus();
        ManosabaSounds.SOUND_EVENTS.register(modEventBus);
        MinecraftForge.EVENT_BUS.register(this);

        if (FMLEnvironment.dist == Dist.CLIENT) {
            MinecraftForge.registerConfigScreen(ManosabaConfigScreen::new);
            LOGGER.info("Manosaba title screen loaded (Forge 1.20.1).");
        }
    }
}

File: src/main/kotlin/me/shiiyuko/manosaba/Manosaba.kt
package me.shiiyuko.manosaba

import net.minecraft.core.registries.BuiltInRegistries
import net.minecraft.core.registries.Registries
import net.minecraft.resources.ResourceKey
import net.minecraft.resources.ResourceLocation
import net.minecraft.sounds.SoundEvent
import net.neoforged.bus.api.IEventBus
import net.neoforged.fml.common.Mod
import net.neoforged.neoforge.registries.DeferredRegister
import org.slf4j.LoggerFactory

/**
 * Manosaba NeoForge entry point.
 * Replaces the vanilla title screen and splash overlay with 魔法少女ノ魔女裁判 themed UI.
 */
@Mod(Manosaba.MOD_ID)
class Manosaba(modBus: IEventBus) {

    companion object {
        const val MOD_ID = "manosaba"
        private val logger = LoggerFactory.getLogger(MOD_ID)

        /** ResourceLocation for the title BGM sound event */
        val TITLE_MUSIC_ID: ResourceLocation = ResourceLocation.fromNamespaceAndPath(MOD_ID, "music")

        /** DeferredRegister for custom sound events */
        val SOUND_EVENTS: DeferredRegister<SoundEvent> =
            DeferredRegister.create(Registries.SOUND_EVENT, MOD_ID)

        /** Title screen BGM sound event - stored as ResourceKey for lookup */
        private val TITLE_MUSIC_KEY: ResourceKey<SoundEvent> =
            ResourceKey.create(Registries.SOUND_EVENT, TITLE_MUSIC_ID)

        /** Look up the registered sound event. Returns null before registration completes. */
        fun getTitleMusic(): SoundEvent? =
            BuiltInRegistries.SOUND_EVENT.get(TITLE_MUSIC_KEY)
    }

    init {
        SOUND_EVENTS.register(modBus)
        logger.info("Manosaba (NeoForge) initialized!")
    }
}
